packages/components/src/DateTimePickers/DateTime/Manager/Manager.component.js

```diff
@@ -103,12 +94,8 @@ class ContextualManager extends React.Component {
 	}
 
 	onChange(event, origin) {
-		const { errorMessage, datetime, textInput, errors, previousErrors } = this.state;
-
-		if (
-			this.props.onChange &&
-			(this.dateHasChanged() || haveErrorsChanged(previousErrors, errors))
-		) {
+		const { errorMessage, datetime, textInput, errors } = this.state;
```

Comment:
You can move that in the if.
Or better, make a fast return instead `if (!this.props.onChange) { return; } // then add the onChange code`